Blog: post about automation rules v2 (#405)
readthedocs/site-community: content/posts/automation-rules-v2.md

title: Automation rules: webhook filters and more flexible matching

> date: 2026-05-26

Author: ```suggestion
date: 2026-06-01
```

> date: 2026-06-01

Author: ```suggestion
date: 2026-06-02
```

title: Automation Rules v2: webhook filters and more flexible matching
date: 2026-05-05
description: Automation rules are now more powerful — match multiple version types in a single rule, filter builds by changed files, commit message or pull request labels, and trigger builds on demand.
category: Feature announcement
tags: hosting, automation, github
authors: Manuel Kaufmann
status: published
image: /images/automation-rules-v2.jpg

> image_credit: Photo by <a href="https://unsplash.com/@jontyson?utm_content=creditCopyText&utm_medium=referral&utm_source=unsplash">Jon Tyson</a> on <a href="https://unsplash.com/photos/white-and-black-no-smoking-sign-Om2C3IYmG6o?utm_content=creditCopyText&utm_medium=referral&utm_source=unsplash">Unsplash</a>

Reviewer: I think maybe a screenshot would be better here? This is an intense image.

Author: I'm happy to change the image.

I'm not sure about a screenshot here, what do you have in mind? The feature it's just a form 🙃 

<img width="950" height="995" alt="Screenshot_2026-05-13_16-43-06" src="https://github.com/user-attachments/assets/6ff81062-3be9-410a-99a6-2743105bf6f8" />

Reviewer: Heh... yea, not ideal :/ Honestly not sure -- I thinkt he webhook files match pattern is useful.

[Automation rules](https://docs.readthedocs.com/platform/stable/automation-rules.html) let project maintainers
automate actions on new branches and tags pushed to their Git repository:
activate a version, hide it, set it as default, delete it on branch deletion, and more.
Today we are announcing **Automation Rules v2**,

> a redesign that makes rules more flexible and unlocks a long-requested capability:

Reviewer 2: I would basically never use v1/v2/etc when describing something to customers. v1/v2 are already poor descriptors for our internal use, but they definitely lose meaning outside core team.

```suggestion
Today we are announcing several new features to automation rules that will make rules more flexible and unlock a long-requested capability:
```

**filtering builds based on the contents of a webhook event**.

## What's new

### Match multiple version types in a single rule

Previously, a rule applied to either a tag *or* a branch.
Now a rule can target any combination of **tags**, **branches**, and **pull requests** at the same time.
This avoids duplicating the same rule with different version types just to cover all the cases.

### Webhook filters

Rules can now look beyond the version name.
When a push or pull request webhook is received,
a rule can also match against:

- **Changed files** — using `fnmatch` patterns (e.g. `docs/*.rst`, `.readthedocs.yaml`).
- **Commit message** — using a Python regular expression (e.g. `^docs:`).
- **Pull request labels** — using a Python regular expression (e.g. `^docs|build$`).

> 

Author: @copilot apply changes based on [this feedback](https://github.com/readthedocs/website/pull/405#discussion_r3203003192)

> 

Author: We need to fix this in the code as well.

All the configured filters must match for the rule to fire.

### New "Trigger build" action

Combined with webhook filters, the new **Trigger build for version** action lets you
**only build when something relevant changed**.
A few use cases this enables:

- Skip builds for commits that don't touch the docs (e.g. only build when files under `docs/` change).
- Skip builds for commits whose message contains `[skip ci]`.
- Only build pull requests that carry a `documentation` label.

When a project has at least one enabled "Trigger build" rule,
builds are gated by those rules:
if no rule matches the incoming webhook event, the build is skipped.

### Enable/disable a rule without deleting it

Each rule now has an **Enabled** toggle,
so you can temporarily turn a rule off without losing its configuration.

## Where it works

Webhook filters rely on the data sent by your Git provider.
For now, they are supported on projects connected through our

> [GitHub App integration](https://docs.readthedocs.com/platform/stable/reference/git-integration.html).

Reviewer: We don't support normal git webhooks? That's a pretty huge limitation.

Author: No, we didn't implemented it for OAuth connections. We could if we want, but I think we want to use this as an excuse for people to migrate from OAuth since it's been deprecated.

Reviewer: Sure, but for non-GitHub users, they don't have a choice sadly. But we should make that more visible in the copy.

Version-only rules (the existing "Activate version", "Hide version", etc.) keep working on every provider as before.

## Migration

If you already have automation rules configured, **there is nothing you need to do**.
All your existing rules have been migrated to the new model and continue to work the same way.
You will simply find new fields available the next time you edit them.

## Try it out

> Head over to your project's **Settings → Automation Rules** page to take it for a spin,

Reviewer 2: ```suggestion
Head over to your project's **Settings → Automation rules** page to take it for a spin,
```

> Head over to your project's :guilabel:`Settings → Automation Rules` page to take it for a spin,

Author: @copilot apply changes based on [this feedback](https://github.com/readthedocs/website/pull/405#discussion_r3203003077)

> and check our [Automation Rules documentation](https://docs.readthedocs.com/platform/stable/automation-rules.html)

Reviewer 2: ```suggestion
and check our [documentation on automation rules](https://docs.readthedocs.com/platform/stable/automation-rules.html)
```
